### PR TITLE
Adsr gradient

### PR DESCRIPTION
**Pull request by jorshi**
Adjusted the calculation in the ADSR to get the gradients working. I guess torch linspace and reversing the signal for the decay and release was confusing the autograd? Computing the curves slightly more directly seems to be working. This includes examples that @turian did in PR #57

### CODE REVIEW

File: src/ddspdrum/torchmodule.py [hunk 1]
```diff
@@ -274,30 +279,27 @@ def attack(self):
...
-        if num_samples < len(out_):
+        if num_samples <= len(out_):
             out_ = out_[:num_samples]
-        elif num_samples > len(out_):
+        else:
             hold_samples = num_samples - len(out_)
-            assert hold_samples.ndim == 0
-            out_ = torch.nn.functional.pad(
-                out_, [0, hold_samples], value=out_[-1].item()
-            )
```
Comment (maxsolomonhenry) [inline src/ddspdrum/torchmodule.py:300]: (╯°□°)╯︵ ┻━┻

File: src/ddspdrum/torchmodule.py [hunk 2]
```diff
@@ -274,30 +279,27 @@ def attack(self):
...
+            sustain = torch.ones(hold_samples) * self.p("sustain")
+            out_ = torch.cat((out_, sustain))
+
```
Comment (maxsolomonhenry) [inline src/ddspdrum/torchmodule.py:302]: ┬─┬ノ( º _ ºノ)

Comment (jorshi) [reply src/ddspdrum/torchmodule.py:302]: hahaha

**Review by turian — APPROVED** [review]
